src/core/Akka/Actor/ActorPath.cs

```diff
@@ -327,74 +327,165 @@ public static bool TryParse(string path, out ActorPath actorPath)
+                }
+
+                spanified = spanified.Slice(nextSlash + 1);
+            } while (nextSlash >= 0);
+
+            //var fragmentLocation = 
+            //if (.StartsWith("#"))
+            //{
+            //    var uid = SpanHacks.Parse(uri.Fragment.AsSpan(1));
+            //    actorPath = actorPath.WithUid(uid);
+            //}
             return true;
         }
 
         /// <summary>
-        /// TBD
+        /// Attempts to parse an <see cref="Address"/> from a stringified <see cref="ActorPath"/>.
         /// </summary>
-        /// <param name="path">TBD</param>
-        /// <param name="address">TBD</param>
-        /// <returns>TBD</returns>
+        /// <param name="path">The string representation of the <see cref="ActorPath"/>.</param>
+        /// <param name="address">If <c>true</c>, the parsed <see cref="Address"/>. Otherwise <c>null</c>.</param>
+        /// <returns><c>true</c> if the <see cref="Address"/> could be parsed, <c>false</c> otherwise.</returns>
         public static bool TryParseAddress(string path, out Address address)
         {
-            Uri uri;
-            return TryParseAddress(path, out address, out uri);
+            return TryParseAddress(path, out address, out var _);
         }
 
-        private static bool TryParseAddress(string path, out Address address, out Uri uri)
+        /// <summary>
+        /// Attempts to parse an <see cref="Address"/> from a stringified <see cref="ActorPath"/>.
+        /// </summary>
+        /// <param name="path">The string representation of the <see cref="ActorPath"/>.</param>
+        /// <param name="address">If <c>true</c>, the parsed <see cref="Address"/>. Otherwise <c>null</c>.</param>
+        /// <param name="absoluteUri">A <see cref="ReadOnlySpan{T}"/> containing the path following the address.</param>
+        /// <returns><c>true</c> if the <see cref="Address"/> could be parsed, <c>false</c> otherwise.</returns>
+        private static bool TryParseAddress(string path, out Address address, out ReadOnlySpan<char> absoluteUri)
         {
-            //This code corresponds to AddressFromURIString.unapply
             address = null;
-            if (!Uri.TryCreate(path, UriKind.Absolute, out uri))
+
+            var spanified = path.AsSpan();
+            absoluteUri = spanified;
+
+            var firstColonPos = spanified.IndexOf(':');
+
+            if (firstColonPos == -1) // not an absolute Uri
                 return false;
-            var protocol = uri.Scheme; //Typically "akka"
-            if (!protocol.StartsWith("akka", StringComparison.OrdinalIgnoreCase))
-            {
-                // Protocol must start with 'akka.*
+
+            var fullScheme = SpanHacks.ToLowerInvariant(spanified.Slice(0, firstColonPos));
+            if (!fullScheme.StartsWith("akka"))
+                return false;
+
+            spanified = spanified.Slice(firstColonPos + 1);
+            if (!(spanified[0] == '/' && spanified[1] == '/'))
```

Comment:
Could throw here on a malformed URI on either of these last two lines; may want to consider:

 - For the Slice:
   - Maybe a length check with the scheme check?  `if (!fullScheme.StartsWith("akka") || spanified.Length < firstColonPos + 2)` (Double check my math here but I think you get the idea)

 - For the If statement: 
   - `if (spanified.Length<2 || !(spanified[0] == '/' && spanified[1] == '/'))`

Reply:
Got it - I'll add that